### PR TITLE
ss2: make an extra

### PR DESCRIPTION
**Pull request by svinota**
Declare psutil in `extras_require` and raise an exception if it is
not present in the system

Bug-Url: https://github.com/svinota/pyroute2/pull/679

### CODE REVIEW

**Comment by svinota** [comment]
@ffourcot @jonringer 

please review this change before I proceed — here I change distutils to setuptools by default, and it may affect

File: cli/ss2 [hunk 1]
```diff
@@ -578,7 +584,7 @@ def run(args=None):
     with DiagSocket() as ds:
         ds.bind()
         for p in protocols:
-            p(ds, args, _user_ctxt_map)
+                p(ds, args, _user_ctxt_map)
```
Comment (cherusk) [inline cli/ss2]: lapsus!

**Comment by cherusk** [comment]
Should we also mention it in some readme local to  cli/, but optional.

File: cli/ss2 (hunk 1 above)
Comment (svinota) [reply cli/ss2]: indeed

Comment (jonringer) [reply cli/ss2]: this looks over-indented. You use 4-wide everywhere, but 8 here

File: setup.py [hunk 2]
```diff
@@ -66,6 +66,7 @@
                 'pyroute2.remote'],
       scripts=['./cli/ss2',
                './cli/pyroute2-cli'],
+      extras_require={'ss2': ['psutil>=5.0']},
```
Comment (jonringer) [inline setup.py]:
```suggestion
      extras_require={'ss2': ['psutil~=5.0']},
```
you will still want an upperbound, as 6.0 may break your use cases (~=5.0 -> >=5,<6)

File: cli/ss2 (hunk 1 above)
Comment (svinota) [reply cli/ss2]: fixed

**Pull request merged**